site/src/App/guides/testing-guide.tsx

```diff
@@ -0,0 +1,251 @@
+        Instead of being resilient to refactors, they break even when
+        meaningless changes occur (e.g. class names change).
+      </Text>
+
+      <Text>
+        What&apos;s often raised as a solution to some of the above is shallow
+        rendered snapshots. Check out &quot;
+        <ExternalLink href="https://kentcdodds.com/blog/why-i-never-use-shallow-rendering">
+          Why I Never Use Shallow Rendering
+        </ExternalLink>
+        &quot; for great article on the issues with shallow rendering.
+      </Text>
+
+      <Text>
+        Note: The above is specifically discussing react component snapshots.
+        Snapshot testing is a great tool when applied to the appropriate
+        use-case.
+      </Text>
+
+      <Heading level="3">What about integration tests?</Heading>
+
+      <Text>
+        We suggest all teams have at least a small number of integration tests
+        making sure their core flows work as expected. Here&apos;s a couple of
+        great libraries to consider:
+      </Text>
+
+      <BulletList>
+        <Bullet>
+          <ExternalLink href="https://www.cypress.io/">Cypress</ExternalLink>
+        </Bullet>
+        <Bullet>
+          <ExternalLink href="https://testing-library.com/docs/cypress-testing-library/intro">
+            Cypress Testing Library
+          </ExternalLink>
+        </Bullet>
+        <Bullet>
+          <ExternalLink href="https://github.com/puppeteer/puppeteer">
+            Puppeteer
+          </ExternalLink>
+        </Bullet>
+        <Bullet>
+          <ExternalLink href="https://github.com/smooth-code/jest-puppeteer">
+            jest-puppeteer
+          </ExternalLink>
+        </Bullet>
+        <Bullet>
+          <ExternalLink href="https://testing-library.com/docs/pptr-testing-library/intro">
+            Puppeteer Testing Library
+          </ExternalLink>
+        </Bullet>
+      </BulletList>
+
+      <Heading level="3">
+        How do I query for elements rendered by Braid?
+      </Heading>
+
+      <Text>
+        If you need to query for an element rendered via a Braid component we
+        suggest using{' '}
+        <ExternalLink href="https://testing-library.com/docs/dom-testing-library/api-queries#byrole">
+          role based querying
+        </ExternalLink>
+        . Targetting labels and ids is also a good option. Some Braid components
+        also allow passing{' '}
+        <ExternalLink href="https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes">
+          html data attributes
+        </ExternalLink>{' '}
+        via the `data` prop. This allows you to pass the `data-testid` prop used
+        by React Testing Library.
+      </Text>
+
+      <Text>
+        If you are having trouble targetting an element in a test please reach
+        out to us in{' '}
+        <ExternalLink href="https://seekchat.slack.com/channels/braid-support">
+          #braid-support
```

Comment:
Since these documents are publicly accessible and this is an open source project, is it appropriate to provide links to our internal SEEK Slack?